//
// Copyright 2016-2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
syntax = "proto2";

package magritte;

// A message encapsulating similarity transform of a sprite.
message SpritePose {
  // Position of the sprite in the background (normalized).
  optional float position_x = 1 [default = 0.5];
  optional float position_y = 2 [default = 0.5];

  // The orientation of the sprite in radians, 0 is upright, positive values
  // correspond to clockwise rotation.
  optional float rotation_radians = 3 [default = 0.0];

  // The scale applied to the sprite. This is multiplied by the sprite
  // dimensions (in pixels) to obtain the desired size.
  optional float scale = 4 [default = 1.0];
}
